fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/MultiMatch.java

```diff
@@ -24,41 +24,46 @@
 import org.apache.doris.nereids.trees.expressions.shape.BinaryExpression;
 import org.apache.doris.nereids.trees.expressions.visitor.ExpressionVisitor;
 import org.apache.doris.nereids.types.BooleanType;
-import org.apache.doris.nereids.types.StringType;
+import org.apache.doris.nereids.types.coercion.AnyDataType;
 
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 
 import java.util.List;
 
 /**
- * ScalarFunction 'multi_match'. This class is generated by GenerateFunction.
+ * ScalarFunction 'multi_match'.
  */
 public class MultiMatch extends ScalarFunction
         implements BinaryExpression, ExplicitlyCastableSignature, AlwaysNotNullable {
 
     public static final List<FunctionSignature> SIGNATURES = ImmutableList.of(
             FunctionSignature.ret(BooleanType.INSTANCE)
-                    .args(StringType.INSTANCE,
-                            StringType.INSTANCE,
-                            StringType.INSTANCE,
-                            StringType.INSTANCE)
+                    .varArgs(AnyDataType.INSTANCE_WITHOUT_INDEX)
```

Comment:
why could support any type? how to do match for any non-string type? i think should be `varArgs(StringType.INSTANCE)` ?

Reply:
changed to varArgs(StringType.INSTANCE)